pkg/webui/console/components/mac-settings-section/index.js

```diff
@@ -0,0 +1,194 @@
+}))
+
+// 0...7
+const pingSlotPeriodicityOptions = Array.from({ length: 8 }, (_, index) => {
+  const value = Math.pow(2, index)
+
+  return {
+    value: `PING_EVERY_${value}S`,
+    label: <Message content={m.pingSlotPeriodicityValue} values={{ count: value }} />,
+  }
+})
+
+// 0...7
+const dataRateOffsetOptions = Array.from({ length: 8 }, (_, index) => ({
+  value: index,
+  label: index,
+}))
+
+const MacSettingsSection = props => {
+  const { activationMode, deviceClasses, resetsFCnt: initialFCnt, initiallyCollapsed } = props
+
+  const isClassB = deviceClasses.includes(DEVICE_CLASSES.CLASS_B)
+  const isOTAA = activationMode === ACTIVATION_MODES.OTAA
+  const isABP = activationMode === ACTIVATION_MODES.ABP
+
+  const [resetsFCnt, setResetsFCnt] = React.useState(isABP && initialFCnt)
+  const handleResetsFCntChange = React.useCallback(evt => {
+    const { checked } = evt.target
+
+    setResetsFCnt(checked)
+  }, [])
+
+  return (
+    <Form.CollapseSection
+      id="mac-settings"
+      title={m.macSettings}
+      initiallyCollapsed={initiallyCollapsed}
+    >
+      {isABP && (
+        <Form.Field
+          title={m.rx1DelayTitle}
+          description={m.rx1DelayDescription}
+          name="mac_settings.rx1_delay.value"
+          component={Select}
+          options={rx1DelayOptions}
+        />
+      )}
+      {(isABP || isOTAA) && (
+        <Form.Field
+          title={m.rx1DataRateOffsetTitle}
+          name="mac_settings.rx1_data_rate_offset"
+          component={Select}
+          options={dataRateOffsetOptions}
+        />
+      )}
+      {isABP && (
+        <Form.Field
+          title={m.resetsFCnt}
+          onChange={handleResetsFCntChange}
+          warning={resetsFCnt ? m.resetWarning : undefined}
+          name="mac_settings.resets_f_cnt"
+          component={Checkbox}
+        />
+      )}
+      <Form.Field
+        title={m.rx2DataRateIndexTitle}
+        description={m.rx2DataDateIndexDescription}
+        name="mac_settings.rx2_data_rate_index.value"
+        component={Select}
+        options={dataRateIndexOptions}
+      />
+      <Form.Field
+        type="number"
+        min={100000}
+        step={100}
+        title={m.rx2FrequencyTitle}
+        description={m.rx2FrequencyDescription}
+        placeholder={m.frequencyPlaceholder}
+        name="mac_settings.rx2_frequency"
+        component={Input}
+      />
+      <Form.Field
+        indexAsKey
+        name="mac_settings.factory_preset_frequencies"
+        component={KeyValueMap}
+        title={m.factoryPresetFreqTitle}
+        description={m.factoryPresetFreqDescription}
+        addMessage={m.freqAdd}
+        valuePlaceholder={m.frequencyPlaceholder}
+      />
+      {isClassB && (
+        <>
+          <Form.Field
+            title={m.pingSlotPeriodicityTitle}
+            description={m.pingSlotPeriodicityDescription}
+            name="mac_settings.ping_slot_periodicity.value"
+            component={Select}
```

Comment:
What else?

Reply:
I'm not familiar with this setting but in depth, but if the ping slot periodicity allows for a range of numbers, we should rather use the `type="number"` input.

Reply:
`ping_slot_periodicity` enum values can be mapped to a numeric value (see https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.9/api/api.md#enum-pingslotperiod). but is it really more convenient for the user to specify `0` to set the period to 1s, instead of just choosing `every second` in the select?

Also related https://github.com/TheThingsNetwork/lorawan-stack/issues/2258

Reply:
Ok yeah, I overlooked how the value is composed and `<Select />` is indeed the way to go here.